examples/2_read_serial.py

```diff
@@ -37,8 +37,8 @@
     print("Field E.x has shape {0} and datatype {1}".format(
           shape, E_x.dtype))
 
-    offset = openPMD.Extent([1, 1, 1])
-    extent = openPMD.Extent([2, 2, 1])
+    offset = [1, 1, 1]
```

Comment:
Will any Python object that is not cleanly castable to a ```std::vector< uint64_t >``` raise?
What about 
```python
offset = [-1, -1, -1]
```
or
```python
offset = 1
```
?

Reply:
I will just try it and report back :)

Reply:
The error is mean, but yes it throws as it should:
```python
offset = [-1, -1, -1]
E_x.load_chunk(offset, extent)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-11-31a35ce5d5ec> in <module>()
----> 1 E_x.load_chunk(offset, extent)

TypeError: load_chunk(): incompatible function arguments. The following argument types are supported:
    1. (self: openPMD.Record_Component, arg0: List[int], arg1: List[int]) -> array

Invoked with: <openPMD.Mesh_Record_Component of dimensionality '3'>, [-1, -1, -1], [2, 2, 1]
```

We just need do add a better docstring (and docstrings at all) to the Python APIs (args).

## Wrong Dimension

```
offset = [1, 1]

In [15]: E_x.load_chunk(offset, extent)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-15-31a35ce5d5ec> in <module>()
----> 1 E_x.load_chunk(offset, extent)

RuntimeError: Dimensionality of chunk and dataset do not match.

In [20]: offset = 1

In [21]: E_x.load_chunk(offset, extent)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-21-31a35ce5d5ec> in <module>()
----> 1 E_x.load_chunk(offset, extent)

TypeError: load_chunk(): incompatible function arguments. The following argument types are supported:
    1. (self: openPMD.Record_Component, arg0: List[int], arg1: List[int]) -> array

Invoked with: <openPMD.Mesh_Record_Component of dimensionality '3'>, 1, [2, 2, 1]
```

### Numpy Indexes

I could allow that with an interface change, but question is if that's a good idea. Maybe
```python
import numpy as np

x = np.ones(2, dtype=np.int)

In [14]: a = [[1.,2.],[3.,4.]]

In [15]: b = np.array(a)

In [16]: b[x]
Out[16]: 
array([[3., 4.],
       [3., 4.]])

In [17]: a[x]
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-17-8fab1da9f470> in <module>()
----> 1 a[x]

TypeError: only integer scalar arrays can be converted to a scalar index
```

```python
offset = np.ones(3)

In [19]: E_x.load_chunk(offset, extent)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-19-31a35ce5d5ec> in <module>()
----> 1 E_x.load_chunk(offset, extent)

TypeError: load_chunk(): incompatible function arguments. The following argument types are supported:
    1. (self: openPMD.Record_Component, arg0: List[int], arg1: List[int]) -> array

Invoked with: <openPMD.Mesh_Record_Component of dimensionality '3'>, array([1., 1., 1.]), [2, 2, 1]
```